datadog_checks_base/datadog_checks/base/stubs/aggregator.py

```diff
@@ -301,6 +301,45 @@ def assert_all_metrics_covered(self):
             msg += '\nMissing Metrics:{}{}'.format(prefix, prefix.join(sorted(self.not_asserted())))
         assert condition, msg
 
+    def assert_metrics_using_metadata(self, metadata_metrics, check_metric_type=True, exclude=None):
```

Comment:
wdyt? since more integrations would have integration tests and not e2e

```suggestion
    def assert_metrics_using_metadata(self, metadata_metrics, check_metric_type=False, exclude=None):
```

Reply:
@ofek 
I think this method has the most value for e2e since check the type is quite valuable. 

Having `check_metric_type=True` by default will encourage that. WDYT?